components/centraldashboard/public/components/main-page.js

```diff
@@ -205,6 +215,36 @@ export class MainPage extends PolymerElement {
     _isInsideOfIframe() {
         return window.location !== window.parent.location;
     }
+
+    /**
+     * Builds and returns an href value preserving the current query string.
+     * @param {string} href
+     * @param {Object} queryParams
+     * @return {string}
+     */
+    _buildHref(href, queryParams) {
+        const url = new URL(href, window.location.origin);
+        VALID_QUERY_PARAMS.forEach((qp) => {
+            if (queryParams[qp]) {
+                url.searchParams.set(qp, queryParams[qp]);
+            }
+        });
+        return url.href.slice(url.origin.length);
+    }
+
+    /**
+     * Sends a message to the iframe message bus. This is used on the iframe
+     * load event as well as when the namespace changes.
+     */
+    _sendNamespaceMessage() {
+        const iframe = this.$.PageFrame;
+        if (this.page === 'iframe' && this.queryParams.ns && iframe.src) {
```

Comment:
_**[nit]**_ Decrease code depth
```suggestion
        if (!(this.page === 'iframe' && this.queryParams.ns && iframe.src)) return;
```